src/sidebar/reducers/annotations.js

```diff
@@ -262,22 +261,21 @@ function addAnnotations(annotations, now) {
     // successfully anchor then the status will be updated.
     var ANCHORING_TIMEOUT = 500;
 
-    var anchoringAnnots = added.filter(metadata.isWaitingToAnchor);
-    if (anchoringAnnots.length) {
-      setTimeout(function () {
-        arrayUtil
-          .filterMap(anchoringAnnots, function (annot) {
-            return findByID(getState().annotations, annot.id);
-          })
-          .filter(metadata.isWaitingToAnchor)
-          .forEach(function (orphan) {
-            dispatch({
-              type: actions.UPDATE_ANCHOR_STATUS,
-              anchorTimeout: true,
-              id: orphan.id,
-              tag: orphan.$tag,
-            });
-          });
+    var anchoringIDs = added.filter(metadata.isWaitingToAnchor)
+                            .map(ann => ann.id);
+    if (anchoringIDs.length > 0) {
+      setTimeout(() => {
+        // Find annotations which haven't yet been anchored in the document.
+        var anns = getState().annotations;
+        var annsStillAnchoring = anchoringIDs.map(id => findByID(anns, id))
+                                             .filter(ann => ann && metadata.isWaitingToAnchor(ann));
+
+        // Mark anchoring as timed-out for these annotations.
+        var anchorStatusUpdates = annsStillAnchoring.reduce((updates, ann) => {
+          updates[ann.$tag] = 'timeout';
+          return updates;
+        }, {});
+        dispatch(updateAnchorStatus(anchorStatusUpdates));
```

Comment:
This is responsible for marking anchoring as having timed-out for newly loaded annotations if they fail to anchor within a certain time period. I modified this initially to dispatch _one_ action with N updates instead of N actions with one update each. In a second commit I rewrote this whole section for clarity in response to CR feedback.

Reply:
I see, so this is a function that runs after 500ms and orphans any annotations that haven't anchored yet. It'll end up calling the `UPDATE_ANCHOR_STATUS` function above, which will set `$anchorTimeout: true`. It seems odd that this is in the sidebar code rather than in the annotator code, but I appreciate that you've rewritten it in-place for clarity.

Reply:
The reason it is in the sidebar rather than the annotator code is because the "annotator" code runs in the context of the host page and there are various ways in which that could cause the timeout code to either not run or to fail to execute within the timeout period. For example, a news website bogged down by a ton of crappy JS and ads.

Putting the code in the sidebar avoids some of these failure scenarios and, depending on the browser's JS execution scheduling, avoids its execution being blocked by code in the host page.